crates/sui-keys/src/key_derive.rs

```diff
@@ -110,11 +153,23 @@ pub fn validate_path(
                     "m/{DERVIATION_PATH_PURPOSE_SECP256K1}'/{DERIVATION_PATH_COIN_TYPE}'/0'/0/0"
                 )
                 .parse()
-                .unwrap()),
+                .map_err(|_| SuiError::SignatureKeyGenError("Cannot parse path".to_string()))?),
             }
         }
         SignatureScheme::BLS12381 => Err(SuiError::UnsupportedFeatureError {
-            error: "BLS is not supported for user key derivation".to_string(),
+            error: "BLS12381 key derivation is currently not supported".to_string(),
         }),
     }
 }
+
+pub fn generate_new_key(
+    key_scheme: SignatureScheme,
+    derivation_path: Option<DerivationPath>,
+) -> Result<(SuiAddress, SuiKeyPair, SignatureScheme, String), anyhow::Error> {
+    let mnemonic = Mnemonic::random(OsRng, Default::default());
+    let seed = mnemonic.to_seed("");
+    match derive_key_pair_from_path(seed.as_bytes(), derivation_path, &key_scheme) {
+        Ok((address, kp)) => Ok((address, kp, key_scheme, mnemonic.phrase().to_string())),
+        Err(e) => Err(anyhow!("Failed to generate keypair: {:?}", e)),
+    }
+}
```

Comment:
minor refactor here so that`keytool generate` and `client new-address` can share the same helper function